pycm/generate_random_data.py

```diff
@@ -69,16 +74,20 @@ def _calculate_class_counts(class_percentages, total_population):
     return dict(zip(classes, class_counts.astype(int).tolist()))
 
 
-def generate_confusion_matrix(class_percentages, total_population):
+def generate_confusion_matrix(class_percentages, total_population, seed=None):
     """
     Generate a random confusion matrix with given class percentages and total population.
 
     :param class_percentages: dict or list of percentages for each class (sum should be 100)
     :type class_percentages: dict or list
     :param  total_population: total number of samples in the confusion matrix
     :type total_population: int
+    :param seed: random seed for reproducibility
+    :type seed: int or None
     :return: confusion matrix as a dictionary
     """
+    if seed is not None:
```

Comment:
I recommend eliminating this condition and instead calling `np.random.seed(seed)` directly.